ui/shared/components/panel/view-fields/BaseFieldView.jsx

```diff
@@ -18,76 +20,126 @@ const FieldValue = styled.div`
 
-  const editButton = props.isEditable && (props.formFields ?
-    <UpdateButton
-      key="edit"
-      modalTitle={props.modalTitle}
-      modalId={modalId}
-      modalSize={props.modalSize}
-      buttonText={props.editLabel}
-      editIconName={props.editIconName}
-      onSubmit={props.onSubmit}
-      initialValues={props.initialValues}
-      formFields={props.formFields}
-      formContainer={<div style={props.modalStyle} />}
-      showErrorPanel={props.showErrorPanel}
-      confirmDialog={props.addConfirm}
-      size="tiny"
-    />
-    : (
-      <DispatchRequestButton
-        key="edit"
-        buttonContent={<Icon link size="small" name="plus" />}
-        onSubmit={() => props.onSubmit(props.initialValues)}
-        confirmDialog={props.addConfirm}
+  render() {
+    if (this.props.isVisible !== undefined && !this.props.isVisible) {
+      return null
+    }
+    if (this.props.isPrivate && !this.props.user.isStaff) {
+      return null
+    }
+    const fieldValue = this.props.initialValues[this.props.field]
+    const hasValue = (fieldValue && (!Object.getOwnPropertyNames(fieldValue).includes('length') || fieldValue.length > 0)) || this.props.showEmptyValues
+    if (!this.props.isEditable && !hasValue) {
+      return null
+    }
+    const fieldId = this.props.initialValues[this.props.idField]
+    const modalId = this.props.isEditable ? `edit-${fieldId || 'new'}-${this.props.field}-${this.props.modalId}` : null
+
+    const updateButton = this.props.showInLine ?
+      <div>
+        {this.state.showInLineButton &&
+        <ButtonLink
+          icon="write"
+          onClick={this.toggleButtonVisibility}
+        />}
+        {!this.state.showInLineButton &&
+        <Segment>
+          <ReduxFormWrapper
+            noModal
+            onSubmit={this.props.onSubmit}
+            closeInlineModal={this.toggleButtonVisibility}
+            form={this.props.modalId}
+            initialValues={this.props.initialValues}
+            fields={this.props.formFields}
+            showErrorPanel={this.props.showErrorPanel}
+            confirmDialog={this.props.confirmDialog}
+            confirmCloseIfNotSaved
+          />
+        </Segment>
+        }
+      </div>
+      : (
+        <UpdateButton
+          showInLine={this.props.showInLine}
+          key="edit"
+          modalTitle={this.props.modalTitle}
+          modalId={modalId}
+          modalSize={this.props.modalSize}
+          buttonText={this.props.editLabel}
+          editIconName={this.props.editIconName}
+          onSubmit={this.props.onSubmit}
+          initialValues={this.props.initialValues}
+          formFields={this.props.formFields}
+          formContainer={<div style={this.props.modalStyle} />}
+          showErrorPanel={this.props.showErrorPanel}
+          confirmDialog={this.props.addConfirm}
+          size="tiny"
+        />
+      )
+
+    const editButton = this.props.isEditable && (this.props.formFields ?
```

Comment:
So this is very nitpicky, but the way the control flow here is kind of confusing, because before if it wasn't editable `updateButton` was just never defined, and now you are always defining it and then sometimes just not using it. So for non-editable fields, or fields without `formFields` this is kind of confusing because there are all these unused components being made and its kind of hard to read through and follow the logic for what gets created.  I think rather than defining `updateButton` and then maybe assigning it to `editButton`, you should just initialize the `editButton` variable and then assign it using a series of if statememts:
```
let editButton
if (this.props.isEditable) {
   if (this.props.showInLine) {
     ...
   }
   else if (this.props.formFields) {
      editButton = ...
   } else {
      editButton = ...
   }
}
```